src/components/approval-type/ApprovalType.vue

```diff
@@ -0,0 +1,157 @@
+              label="Court Order Number"
+              :rules="courtOrderNumRules"
+              :disabled="approvalTypeSelected === ApprovalTypes.VIA_REGISTRAR"
+              @input="courtOrderNumberChanged"
+              @update:error="emitValidationError($event)"
+              filled
+            />
+          </v-form>
+          <!-- REGISTRAR radio button -->
+          <v-radio v-if="!isCourtOrderOnly"
+             id="registrar-radio" class="mb-0 pt-2"
+             :label="getRadioText(ApprovalTypes.VIA_REGISTRAR)"
+             :value="ApprovalTypes.VIA_REGISTRAR"
+          />
+        </v-radio-group>
+      </v-col>
+    </v-row>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Emit, Prop } from 'vue-property-decorator'
+import { FormIF } from '@bcrs-shared-components/interfaces'
+
+@Component({})
+export default class ApprovalType extends Vue {
+  // Refs
+  $refs!: Vue['$refs'] & {
+    courtNumRef: FormIF
+  }
+
+  ApprovalTypes = {
+    VIA_COURT_ORDER: 'VIA COURT ORDER',
+    VIA_REGISTRAR: 'VIA REGISTRAR'
+  }
+
+  /** Draft court order number. */
+  @Prop({ default: '' }) readonly draftCourtOrderNumber!: string
+
+  /** Whether approved by the registrar. */
+  @Prop({ default: false }) readonly draftApprovedByRegistrar!: boolean
+
+  /** filing name used in radio options. */
+  @Prop({ default: 'restoration'}) readonly filingType!: string
+
+  /** Show only the court order option; remove via registrar option. */
+  @Prop({ default: false }) readonly isCourtOrderOnly!: boolean
+
+  // Local properties
+  private courtOrderNumber = ''
+  private courtOrderNumRules = []
+  private valid = false
+  private approvalTypeSelected = ''
+
+  /** Called when component is mounted. */
+  mounted (): void {
+    // Copy props to mutable properties
+    if (this.draftApprovedByRegistrar) {
+      this.courtOrderNumber = ''
+      this.approvalTypeSelected = this.ApprovalTypes.VIA_REGISTRAR
+    }
+    if (this.draftCourtOrderNumber) {
+      this.courtOrderNumber = this.draftCourtOrderNumber
+      this.approvalTypeSelected = this.ApprovalTypes.VIA_COURT_ORDER
+    }
+  }
+
+  /** Triggers the form validation. */
+  public validate (): boolean {
+    let status = this.$refs.courtNumRef.validate()
+    this.$emit('emitValid', status)
+    return status
+  }
+
+  private emitValidationError (event: boolean): void {
+    this.$emit('emitValid', !event)
+  }
+
+  @Emit('emitRadioButtonChange')
+  private radioButtonChanged (event: string): void {
+    if (event === this.ApprovalTypes.VIA_REGISTRAR) {
+      Vue.set(this, 'courtOrderNumber', '')
+      Vue.set(this, 'courtOrderNumRules', [])
+    }
+    if (event === this.ApprovalTypes.VIA_COURT_ORDER) {
+      this.setCourtNumberRules()
+    }
+    this.validate()
+  }
+
+  @Emit('emitCourtNumberChange')
+  private courtOrderNumberChanged (event): void {
+    Vue.set(this, 'approvalTypeSelected', this.ApprovalTypes.VIA_COURT_ORDER)
```

Comment:
Why can't this be a simple assignment?

Reply:
My understanding is that simple assignments are not reactive.

Reply:
This PR is water under the bridge, but I'm fairly certain simple assignments _are_ reactive;  we use them all the time.

My understanding is that Vue.set adds a property to an existing reactive object (ie, related to deep reactivity). I think it's _probably_ not needed here.